plugins/CoreVisualizations/JqplotDataGenerator/Chart.php

```diff
@@ -179,4 +208,28 @@ private function getXAxis($index)
         }
         return $axisName;
     }
+
+    private function checkDataAmountConsistency(): void
+    {
+        if ([] === $this->data && [] === $this->dataStates) {
```

Comment:
**Question** — I don't mind this way of checking for empty arrays, is that the way/standard we want to adopt in general? Otherwise I'd just go `empty($this->data)`. The strict check is better, but I suspect we may get a future type warnings as we are not probably in the state of code/data where we can reliably do this?

Reply:
The default values are set to empty arrays and `setAxisYValues` is calling `$this->data[] = ...`. I would expect that to be as reliable as it currently can be :thinking:

Reply:
It was more a question around the style, it might be reliable here but if we adopt this approach more widely in other places it can be less reliable than `empty($array)`. We can discuss this outside of the PR, all good, was just curious.